charts/nautobot/templates/_helpers.tpl

```diff
@@ -169,107 +169,32 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
-        {{- $password | b64dec -}}
-      {{- else -}}
-        {{- required "A Postgres Password is required! Path: .Values.postgresql.auth.password" .Values.postgresql.auth.password -}}
-      {{- end -}}
+      {{- default (printf "%s-db-password" (include "common.names.fullname")) (tpl .Values.postgresql.auth.existingSecret $) -}}
   {{- else if eq .Values.postgresqlha.enabled true -}}
-      {{- if .Values.postgresqlha.postgresql.existingSecret -}}
-        {{- $password := "" -}}
-        {{- $secret := (lookup "v1" "Secret" $.Release.Namespace .Values.postgresqlha.postgresql.existingSecret) -}}
-        {{- if $secret -}}
-          {{- if index $secret.data "postgresql-password" -}}
-            {{- $password = index $secret.data "postgresql-password" -}}
-          {{- else -}}
-            {{- fail (printf "Key 'postgresql-password' not found in secret %s" .Values.postgresqlha.postgresql.existingSecret) -}}
-          {{- end -}}
-        {{- else -}}
-          {{- fail (printf "Existing PostgreSQL-HA secret %s not found!" .Values.postgresqlha.postgresql.existingSecret) -}}
-        {{- end -}}
-        {{- $password | b64dec -}}
-      {{- else -}}
-        {{- required "A Postgres Password is required! Path: .Values.postgresqlha.postgresql.password" .Values.postgresqlha.postgresql.password -}}
-      {{- end -}}
+      {{- default (printf "%s-db-password" (include "common.names.fullname")) (tpl .Values.postgresqlha.auth.existingSecret $) -}}
   {{- else if eq .Values.mariadb.enabled true -}}
-      {{- if .Values.mariadb.auth.existingSecret -}}
-        {{- $password := "" -}}
-        {{- $secret := (lookup "v1" "Secret" $.Release.Namespace .Values.mariadb.auth.existingSecret) -}}
-        {{- if $secret -}}
-          {{- if index $secret.data "mariadb-password" -}}
-            {{- $password = index $secret.data "mariadb-password" -}}
-          {{- else -}}
-            {{- fail (printf "Key 'mariadb-password' not found in secret %s" .Values.mariadb.auth.existingSecret) -}}
-          {{- end -}}
-        {{- else -}}
-          {{- fail (printf "Existing MariaDB secret %s not found!" .Values.mariadb.auth.existingSecret) -}}
-        {{- end -}}
-        {{- $password | b64dec -}}
-      {{- else -}}
-        {{- required "A MariaDB Password is required!. Path: .Values.mariadb.auth.password" .Values.mariadb.auth.password -}}
-      {{- end -}}
-  {{- else if .Values.nautobot.db.password -}}
-    {{- .Values.nautobot.db.password -}}
+      {{- default (printf "%s-db-password" (include "common.names.fullname")) (tpl .Values.mariadb.auth.existingSecret $) -}}
   {{- else -}}
-    {{- fail (printf "You have to configure database credentials.") -}}
+    {{- printf "%s-db-password" (include "common.names.fullname" .) -}}
   {{- end -}}
 {{- end -}}
 
-{{- define "nautobot.database.encryptedPassword" -}}
-  {{- include "nautobot.database.rawPassword" . | b64enc | quote -}}
+{{- define "nautobot.database.secretKey" -}}
```

Comment:
Can we use something more explicit than `secretKey`? When I look at this, it seems to me that we are defining the Django secret key. Perhaps something like `passwordKey` or something similar.

Reply:
Changed it to `passwordName` and `passwordKey`.